### ISSUE TITLE
HTTP 500 error csv-importer

### ISSUE THREAD

**Issue by planck0** [firefly-iii/firefly-iii#4722]
**Bug description**
I am running Firefly III version 5.5.9, and CSV import tool version 2.5.1  my problem is: I can't get it to authenticate anymore. It did work with Firefly 5.4.6 an CSV import tool version 2.2.3

When I try to login with personal access token, I get 

''You're using a valid Personal Access Token to authenticate to Firefly III.

Error accessing "https://mydomain.com/firefly/api/v1/about". Status code is 401. Body is: {"message":"Unauthenticated","exception":"AuthenticationException"}''

When I try to login with client id, or without config, I get redirected to my firefly installation and get to the 'Firefly III v5.5.9 Authorization Request'. When I click 'Authorize', I get a HTTP 500 error on my csv-import callback. In the logs I find:

''[2021-04-28 23:04:19] local.DEBUG: Now at App\Http\Controllers\TokenController::callback
[2021-04-28 23:04:19] local.DEBUG: State is valid!
[2021-04-28 23:04:19] local.DEBUG: Params for access token {"form_params":{"grant_type":"authorization_code","client_id":1,"redirect_uri":"https://mydomain.com/csv-importer/callback","code_verifier":"bMqnPdJzy2BZYRG6UMbifuKyTQf0wnVkyJVQu5expbhvPQCtEhexgBU9kPqLh3usbnDo2w0FfqvOf8OiTh2VBzQNSlx0UBAwb72wXIJiYgPsgb8w9AnriNwVEK02ojtw","code":"def502000dcfbbdc50e61666f07e04f58219fa7d38a00eb0daf851bf323abbe3a9f70a96e406c32b38e55aecfb5d49270c60a75cec29fdad9e54ef13f4dd75d2dfc962be1a585d3754f011bda98192676682ac2a03a89d80fa270cd12ad7610f3131223a0dfc9f29b46b6bcf9f840178142716562bc340c5bdf4f1c54bd704bbf7f6e51a9ad1788a43f647bd2ca6df4d39aa16d7073530649d796186a4dabc511ecb3eab3ecb9eeda4f9714bdd3ca5aee4b5e6907bd20d995ec3636291f293f5092a8a2802f2fabc23c665545c713f9481578a58bd4555e574073226090d2840d6c4af1306a9a04103a9da614a344e4423da82ebfe6a4596144c837279832a9145b19808a02fcc6ebe8e3a83c3b1065af55720aaa1339fc9893232225c8b2fc6e0f0bf0c859d0f3bfb9faf5f2522494e59295c7872614629ddf03c0590758118d99259e136705a2ab1c51ab98f55afce129f81a2d096d15f644f5f5c035c3a7c50dd5ae6e4e979cbbb30e7859fe6b07fe7d3be006c516974a7d5eae93f8ba9494834b06c5f8d10648c4926bd6bc9a3c7453b500da336aa3525643b4680cf40"}}
[2021-04-28 23:04:19] local.DEBUG: Will contact "https://mydomain.com/firefly/oauth/token" for a token.
[2021-04-28 23:04:19] local.ERROR: Client error: `POST https://mydomain.com/firefly/oauth/token` resulted in a `401 Unauthorized` response:
{"error":"invalid_client","error_description":"Client authentication failed","message":"Client authentication failed"}
 {"exception":"[object] (GuzzleHttp\\Exception\\ClientException(code: 401): Client error: `POST https://mydomain.com/firefly/oauth/token` resulted in a `401 Unauthorized` response:
{\"error\":\"invalid_client\",\"error_description\":\"Client authentication failed\",\"message\":\"Client authentication failed\"}
 at /var/www/csv-importer/vendor/guzzlehttp/guzzle/src/Exception/RequestException.php:113)''

**Steps to reproduce**
Configure firefly with oath client and personal token, and install new version of csv importer and run it.

**Extra info**
Both Firefly and CSV import  runs on an apache server running Ubuntu 20.04. I tested on a windows client running both firefox and chrome.

**Comment by JC5** [comment]
OK, so no matter what you do, you get 401 errors, is that right?

**Comment by planck0** [comment]
Yes, exactly. Any ideas?

**Comment by JC5** [comment]
What's the result of `php artisan passport:keys`?

**Comment by planck0** [comment]
I get: ''Encryption keys already exist. Use the --force option to overwrite them.''

**Comment by JC5** [comment]
Alright. You could use --force and overwrite them, then create new API keys. Just to see if it works. But honestly this is a pretty foolproof system and when it borks, it's usually PEBKAC. I'm not sure what other steps to take here.

**Comment by planck0** [comment]
I tried that, but it didn't work. I tried some other things:
- I reinstalled version 2.2.3. That one works with personal access token. Didn't try client id method.
- I then reinstalled version 2.5.1 and copied the .env file. That didn't work (both personal access token and client id methods).
- I copied the .env.example file to .env and only altered the personal access token. This works!
- Client id method with 2.5.1 just doesn't work. Maybe I am indeed the problem, but I've no idea what I did wrong. I followed the docs to the letter.

Anyway, I now have something that works. Thanks for your time!
